File: frontend/lib/features/tracking/models/cv_ingredient_item.dart
class CvIngredientItem {
  CvIngredientItem({
    required this.idNguyenLieu,
    required this.tenNguyenLieu,
    required this.tenNguyenLieuKyThuat,
    required this.khoiLuongUocTinhG,
    required this.doChinhXacUocTinh,
  });

  final String idNguyenLieu;
  final String tenNguyenLieu;
  final String tenNguyenLieuKyThuat;
  final double khoiLuongUocTinhG;
  final String doChinhXacUocTinh;

  factory CvIngredientItem.fromJson(Map<String, dynamic> json) {
    return CvIngredientItem(
      idNguyenLieu: (json['id_nguyen_lieu'] ?? json['idNguyenLieu'] ?? '').toString(),
      tenNguyenLieu: (json['ten_nguyen_lieu'] ?? json['tenNguyenLieu'] ?? '').toString(),
      tenNguyenLieuKyThuat:
          (json['ten_nguyen_lieu_ky_thuat'] ?? json['tenNguyenLieuKyThuat'] ?? '').toString(),
      khoiLuongUocTinhG: _number(json['khoi_luong_uoc_tinh_g'] ?? json['khoiLuongUocTinhG']),
      doChinhXacUocTinh: (json['do_chinh_xac_uoc_tinh'] ?? json['doChinhXacUocTinh'] ?? '').toString(),
    );
  }
}

double _number(dynamic value) => value is num ? value.toDouble() : double.tryParse('$value') ?? 0;

File: backend/MenuGreen.BusinessLogicLayer/DTOs/Responses/CvIngredientItem.cs
using System.Text.Json.Serialization;

namespace MenuGreen.BusinessLogicLayer.DTOs.Responses
{
    public class CvIngredientItem
    {
        [JsonPropertyName("id_nguyen_lieu")]
        public string IdNguyenLieu { get; set; } = string.Empty;

        [JsonPropertyName("ten_nguyen_lieu")]
        public string TenNguyenLieu { get; set; } = string.Empty;

        [JsonPropertyName("ten_nguyen_lieu_ky_thuat")]
        public string TenNguyenLieuKyThuat { get; set; } = string.Empty;

        [JsonPropertyName("khoi_luong_uoc_tinh_g")]
        public double KhoiLuongUocTinhG { get; set; }

        [JsonPropertyName("do_chinh_xac_uoc_tinh")]
        public string DoChinhXacUocTinh { get; set; } = string.Empty;
    }
}
